lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
I've made 2 changes:
1. Return the first prototype object and don't go up even if current prototype object is marked as no enumerable property.
2. Remove downward lookup loop.
For the perf benchmark for...in is optimized for, there is a stable score drop (~2%) when comparing with chakra before this change.
Seems skipping the prototype without any enumerable properties is necessary option. For an simple case, for...in on `Array` (this might not be a good pattern, but others are similar) will benefit from this optimization (no need to visit all the properties in Array.prototype and Object.prototype)